src/quickselect.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+/* @flow */
+
+module.exports = quickselect;
+
+/**
+ * Rearrange items in `arr` so that all items in `[left, k]` range are the smallest.
+ * The `k`-th element will have the `(k - left + 1)`-th smallest value in `[left, right]`.
+ *
+ * Implements Floyd-Rivest selection algorithm https://en.wikipedia.org/wiki/Floyd-Rivest_algorithm
+ *
+ * @private
+ * @param {Array<number>} arr input array
+ * @param {number} k pivot index
+ * @param {number} left left index
+ * @param {number} right right index
+ * @returns {undefined}
+ * @example
+ * var arr = [65, 28, 59, 33, 21, 56, 22, 95, 50, 12, 90, 53, 28, 77, 39];
+ * quickselect(arr, 8);
+ * // [39, 28, 28, 33, 21, 12, 22, 50, 53, 56, 59, 65, 90, 77, 95]
+ */
+function quickselect(arr /*: Array<number> */, k /*: number */, left /*: number */, right /*: number */) {
+    left = left || 0;
+    right = right || (arr.length - 1);
+
+    while (right > left) {
+        if (right - left > 600) {
```

Comment:
According to the Wikipedia pseudocode - https://en.wikipedia.org/wiki/Floyd%E2%80%93Rivest_algorithm - 600 is an 'arbitrary' constant' - could we comment / explain this more to make that fact more obvious. The same goes for 0.5